### PR TITLE
Patch 18017 magento 23

### PR DESCRIPTION
**Pull request by niravkrish**
### Description (*)

When pre-selecting a configurable product swatch that has an image, Magento will try to load that image in the product gallery before it is initialized, throwing a JavaScript error.

### Fixed Issues (if relevant)

1. #18017: URL pre-selection of configurable product swatches with associated product images throws JavaScript error

### Manual testing scenarios (*)

1. Hit Url With Url domain.com/montana-wind-jacket.html#93=53
2. Hit Url With Url domain.com/montana-wind-jacket.html#93=58
3. Test with normal configurable product on listing page as well as on Product Details Page

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @niravkrish. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

File: app/code/Magento/Swatches/view/frontend/web/js/swatch-renderer.js [hunk 1]
```diff
@@ -1223,7 +1223,14 @@ define([
                 }
 
                 imagesToUpdate = this._setImageIndex(imagesToUpdate);
-                gallery.updateData(imagesToUpdate);
+                if (gallery === undefined) {
```
Comment (orlangur) [inline app/code/Magento/Swatches/view/frontend/web/js/swatch-renderer.js]: Such check is not correct, please use `typeof` instead.

Comment (niravkrish) [reply app/code/Magento/Swatches/view/frontend/web/js/swatch-renderer.js]: I will update the code.
Please review It.

**Review by VladimirZaets — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @VladimirZaets, thank you for the review.
[ENGCOM-3665](https://jira.corp.magento.com/browse/ENGCOM-3665) has been created to process this Pull Request

**Comment by orlangur** [comment]
@VladimirZaets shouldn't this be `typeof gallery === 'undefined'` instead of `typeof gallery === undefined`?

**Comment by sdzhepa** [comment]
Hello @niravkrish 
During testing our QA engineer faced an issue.
Could you please have a look and fix it?

Please see the quote of QA comment after testing on branch with the PR

PR doesn't fix issue #18017

Manual testing scenario:

1. Create two option for configurable product based on Visual Swatch attribute
2. Set Use Web Server Rewrites=No
3. Open the configurable product page, with the option pre-selected.
` magento_site/index.php/conf1.html#135=4`

**Actual Result:** Not OK,
19635pr




Thank you for contribution and collaboration

**Comment by VladimirZaets** [comment]
@niravkrish, I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for collaboration

**Comment by ghost** [comment]
Hi @niravkrish, thank you for your contribution! </br>Please, complete [Contribution Survey](https://opensource.magento.com/survey/contribution/magento+magento2+19635), it will take less than a minute. </br>Your feedback will help us to improve contribution process.

**Comment by niravkrish** [comment]
Hello,

Please review and test issue.

I have already resolve gallery image change issue in resolve conflicts my last commit ( resolve conflicts - bbceaab ).

**Comment by VladimirZaets** [comment]
Hi, @niravkrish. As I see the merge conflict exists. Please resolve it.

**Comment by VladimirZaets** [comment]
@niravkrish thanks.

**Comment by VasylShvorak** [comment]
✔️ QA passed

**Pull request merged**